src/cirrus/lib/statedb.py

```diff
@@ -139,7 +141,11 @@ def get_items_page(self, collections_workflow: str,
             startkey = { key: dbitem[key] for key in ['collections_workflow', 'itemids', 'state_updated', 'updated']}
             resp = self.query(collections_workflow, state=state, since=since, sort_ascending=sort_ascending, sort_index=sort_index, Limit=limit, ExclusiveStartKey=startkey, )
         else:
-            resp = self.query(collections_workflow, state=state, since=since,  sort_ascending=sort_ascending, sort_index=sort_index, Limit=limit)
+            if error_begins_with:
+                resp = self.query(collections_workflow, error_begins_with=error_begins_with, state=state, since=since,  sort_ascending=sort_ascending, sort_index=sort_index, Limit=limit)
+            else:
+                resp = self.query(collections_workflow, state=state, since=since,  sort_ascending=sort_ascending, sort_index=sort_index, Limit=limit)
```

Comment:
It seems rather than having three places we could run the query operation, we instead refactor this code to have only one call to query, and we build up a query_params dict of all keyword args we want to pass to it. Things like `ExclusiveStartKey` and `error_begins_with` can be added to that dict conditionally.